File: src/main/java/org/cubexmc/metro/train/ArrivalHeuristics.java
package org.cubexmc.metro.train;

final class ArrivalHeuristics {

    private static final double THRESHOLD_BLOCKS = 2.0;

    private ArrivalHeuristics() {
    }

    static boolean shouldArrive(double distanceSquared, double speedBlocksPerTick) {
        double threshold = thresholdBlocks(speedBlocksPerTick);
        return distanceSquared <= threshold * threshold;
    }

    static double thresholdBlocks(double speedBlocksPerTick) {
        return THRESHOLD_BLOCKS;
    }
}

File: src/main/java/org/cubexmc/metro/train/ArrivalHeuristics.kt
package org.cubexmc.metro.train

internal object ArrivalHeuristics {

    private const val THRESHOLD_BLOCKS = 2.0

    @JvmStatic
    fun shouldArrive(distanceSquared: Double, speedBlocksPerTick: Double): Boolean {
        val threshold = thresholdBlocks(speedBlocksPerTick)
        return distanceSquared <= threshold * threshold
    }

    @JvmStatic
    @Suppress("UNUSED_PARAMETER")
    fun thresholdBlocks(speedBlocksPerTick: Double): Double = THRESHOLD_BLOCKS
}
